services/17kmmrbot/src/services/extractor.ts

```diff
@@ -105,32 +109,51 @@ export class ExtractorService {
 		const mmrs =
 			leaderboardEntries?.map((entry) => entry.level_score) ?? [];
 
+		// Check if there are any lords in the lobby in case the user is a spectator
+		const lordLobby: boolean =
+			players
+				.map((player) => player.rank_tier)
+				.filter((rank_tier) => rank_tier === 80).length > 0
+				? true
+				: false;
```

Comment:
You are already getting back a boolean from `.length > 0`

```suggestion
			players
				.map((player) => player.rank_tier)
				.filter((rank_tier) => rank_tier === 80).length > 0;
```

Otherwise a reduce would do too:
```suggestion
			players
				.map((player) => player.rank_tier)
				.reduce((acc, rank_tier) => acc || rank_tier === 80, false);
```